libass/ass.c

```diff
@@ -1235,6 +1235,31 @@ static char *sub_recode(ASS_Library *library, char *data, size_t size,
 }
 #endif                          // ICONV
 
+FILE* ass_fopen(const char* _FileName, const char* _Mode) {
+#if defined(_WIN32) && !defined(__CYGWIN__)
+    wchar_t* wpath, * wmode;
+    UINT cur_cp = CP_UTF8;
+    FILE* f = NULL;
+    wpath = ass_convert_to_utf16le(_FileName, cur_cp), wmode = ass_convert_to_utf16le(_Mode, cur_cp);
+    if (!wpath || !wmode || !(f = _wfopen(wpath, wmode))) {
+        free(wpath), free(wmode);
+        cur_cp = CP_ACP;
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+        if (!AreFileApisANSI()) cur_cp = CP_OEMCP;
+#endif
+        wpath = ass_convert_to_utf16le(_FileName, cur_cp), wmode = ass_convert_to_utf16le(_Mode, cur_cp);
+        if (!wpath || !wmode || !(f = _wfopen(wpath, wmode))) {
```

Comment:
Better to use `fopen()` directly instead of manually emulating it.

Reply:
later commit, too.